autoload/neomake/utils.vim

```diff
@@ -508,6 +508,9 @@ endfunction
 
 function! neomake#utils#write_tempfile(bufnr, temp_file) abort
     call writefile(neomake#utils#get_buffer_lines(a:bufnr), a:temp_file, 'b')
+    if exists('*setfperm')
+        call setfperm(a:temp_file, 'rw-r--r--')
```

Comment:
What about using 600 after all?
Or better: the original file's permissions?

Reply:
I can change it to 600 if you want to. If we want to use the users umask we have to remove this assert that fails:

https://github.com/neomake/neomake/blob/856b272bc09850835e3cd6c47ab5b91b995bd993/tests/tempfiles.vader#L23

Reply:
Yes, the assertion needs to be adjusted in both cases (when changing your PR).

I think using `getfperm(bufname(+a:bufnr))` with a fallback to `'rw-------'` if it is empty would be good.

Reply:
btw: the check for if setfperm exists is good, it is not available in 7.3.429 (the oldest version being tested).  Therefore you need to consider this in the test.

Reply:
We could run the tests with a specific `umask` I guess, which could go into the Makefile then, but I am not sure if it is really worth it (and would need to handle Windows then also (Windows tests are on hold, but there is some WIP in #1575))

Reply:
@NickeZ 
Are you going to update the PR?
Otherwise I might pick it up.

Reply:
Feel free to take it if you have decided what to do. I was waiting for your input.

Reply:
Pushed an update, according to my previous comment(s).

Reply:
The tests succeed on my machine :+1:

Reply:
Thanks, merged.